app/components/project-geometries/types.js

```diff
@@ -38,6 +38,23 @@ export function isMeaningfulChange(geometricPropertyForType /* model */) {
     && isFeatureCollectionChanged(initial, proposed);
 }
 
+export function newPolygonsHaveLabels(geometricPropertyForType) {
+  const [
+    initial,
+    proposed,
+  ] = geometricPropertyForType.changedAttributes().proposedGeometry || [];
+
+  if(proposed) {
+    const polygonWithoutLabel = proposed.features.find((feature) => {
+      return !feature.properties.label; 
+    });
+
+    if(polygonWithoutLabel) return false;
+  }
+
+  return true;
+}
```

Comment:
I see - yes, another PR will move this logic up into the model where it probably belongs. These both are downstream calculations on the geometricProperty model, so it'll fit nicely in `models/geometric-property`.